src/lib/Microsoft.Health.Events/EventHubProcessor/AssignedPartitionProcessor.cs

```diff
@@ -0,0 +1,226 @@
+
+        protected override Task<string[]> ListPartitionIdsAsync(
+            EventHubConnection connection,
+            CancellationToken cancellationToken) =>
+                Task.FromResult(_assignedPartitions);
+
+        // Tell the processor that it owns all of the available partitions for the Event Hub.
+        protected override Task<IEnumerable<EventProcessorPartitionOwnership>> ListOwnershipAsync(
+            CancellationToken cancellationToken) =>
+                Task.FromResult(
+                    _assignedPartitions.Select(partition =>
+                        new EventProcessorPartitionOwnership
+                        {
+                            FullyQualifiedNamespace = FullyQualifiedNamespace,
+                            EventHubName = EventHubName,
+                            ConsumerGroup = ConsumerGroup,
+                            PartitionId = partition.ToString(),
+                            OwnerIdentifier = Identifier,
+                            LastModifiedTime = DateTimeOffset.UtcNow,
+                        }));
+
+        // Accept any ownership claims attempted by the processor; this allows the processor to
+        // simulate renewing ownership so that it continues to own all of its assigned partitions.
+        protected override Task<IEnumerable<EventProcessorPartitionOwnership>> ClaimOwnershipAsync(
+           IEnumerable<EventProcessorPartitionOwnership> desiredOwnership,
+           CancellationToken cancellationToken) =>
+                Task.FromResult(desiredOwnership.Select(ownership =>
+                {
+                    ownership.LastModifiedTime = DateTimeOffset.UtcNow;
+                    return ownership;
+                }));
+
+        protected async override Task OnProcessingEventBatchAsync(
+            IEnumerable<EventData> events,
+            EventProcessorPartition partition,
+            CancellationToken cancellationToken)
+        {
+            try
+            {
+                if (events.Count() == 0)
+                {
+                    var maxWaitEvent = new MaximumWaitEvent(partition.PartitionId, DateTime.UtcNow);
+                    await _eventConsumerService.ConsumeEvent(maxWaitEvent, cancellationToken);
+                    return;
+                }
+
+                foreach (var currentEvent in events)
+                {
+                    var eventMessage = new EventMessage(
+                        partition.PartitionId,
+                        currentEvent.EventBody,
+                        currentEvent.ContentType,
+                        currentEvent.SequenceNumber,
+                        currentEvent.Offset,
+                        currentEvent.EnqueuedTime,
+                        currentEvent.Properties,
+                        currentEvent.SystemProperties);
+
+                    await _eventConsumerService.ConsumeEvent(eventMessage, cancellationToken);
+                }
+            }
+            catch (Exception ex)
```

Comment:
it really should never be hit since we try/catch the exceptions in our handler code and retry if appropriate. the event hub examples also squash it and have the follow comment so I think it is appropriate since we have try/catch and retry logic in the handler anyway. we could also opt to throw if think that crashing and restarting would potentially lead to a better outcome but I am not sure we know

            // It is very important that you always guard against exceptions in
            // your handler code; the processor does not have enough
            // understanding of your code to determine the correct action to take.
            // Any exceptions from your handlers go uncaught by the processor and
            // will NOT be redirected to the error handler.
            //
            // In this case, the partition processing task will fault and be restarted
            // from the last recorded checkpoint.

Console.WriteLine($"Exception while processing events: { ex }");

Reply:
added a trace